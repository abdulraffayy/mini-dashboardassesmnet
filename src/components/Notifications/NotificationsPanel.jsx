import React from 'react';
import { formatDistanceToNow } from 'date-fns';

export default function NotificationsPanel({
  notifications = [],
  onToggleRead = () => { },
  onClearAll = () => { }
}) {
  const unreadCount = notifications.filter(note => !note.read).length;

  const formatTimestamp = (timestamp) => {
    return formatDistanceToNow(new Date(timestamp), { addSuffix: true });
  };

  const notificationIcons = {
    comment: (
      <span className="bg-blue-400 rounded-lg flex items-center justify-center w-10 h-10">
        <svg className="w-6 h-6 text-white" fill="none" viewBox="0 0 24 24"><path d="M21 15a2 2 0 01-2 2H7l-4 4V5a2 2 0 012-2h14a2 2 0 012 2z" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/></svg>
      </span>
    ),
    update: (
      <span className="bg-orange-300 rounded-lg flex items-center justify-center w-10 h-10">
        <svg className="w-6 h-6 text-white" fill="none" viewBox="0 0 24 24"><path d="M12 8v4l3 3" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/><circle cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="2"/></svg>
      </span>
    ),
    password: (
      <span className="bg-emerald-300 rounded-lg flex items-center justify-center w-10 h-10">
        <svg className="w-6 h-6 text-white" fill="none" viewBox="0 0 24 24"><path d="M12 15v2m-6 4h12a2 2 0 002-2V7a2 2 0 00-2-2H6a2 2 0 00-2 2v12a2 2 0 002 2z" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/><path d="M9 11V9a3 3 0 116 0v2" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/></svg>
      </span>
    ),
  };

  function getIcon(type) {
    if (type === 'comment') return notificationIcons.comment;
    if (type === 'update') return notificationIcons.update;
    if (type === 'password') return notificationIcons.password;
    return null;
  }

  const DEFAULT_NOTIFICATIONS = [
    { id: '1', type: 'comment', text: 'New comment', timestamp: new Date(Date.now() - 2 * 60 * 1000) },
    { id: '2', type: 'update', text: 'System update', timestamp: new Date(Date.now() - 60 * 60 * 1000) },
    { id: '3', type: 'password', text: 'Password changed', timestamp: new Date(Date.now() - 5 * 60 * 60 * 1000) },
  ];
  const displayNotifications = notifications && notifications.length > 0 ? notifications : DEFAULT_NOTIFICATIONS;

  return (
    <div className="bg-white p-8 rounded-xl shadow-md">
      <h2 className="text-xl font-semibold mb-6">Notifications</h2>
      {displayNotifications.length === 0 ? (
        <div className="text-center text-gray-500 py-8">
          No notifications
        </div>
      ) : (
        <ul>
          {displayNotifications
            .sort((a, b) => new Date(b.timestamp) - new Date(a.timestamp))
            .map((note, idx, arr) => (
              <li key={note.id}>
                <div className="flex items-start gap-4 pb-4">
                  {getIcon(note.type)}
                  <div className="flex flex-col">
                    <span className="font-bold text-base text-gray-900">{note.text}</span>
                    <span className="text-xs text-gray-500 mt-1">{formatTimestamp(note.timestamp)}</span>
                  </div>
                </div>
                {idx !== arr.length - 1 && <hr className="border-gray-200 mb-4" />}
              </li>
            ))}
        </ul>
      )}
    </div>
  );
}
