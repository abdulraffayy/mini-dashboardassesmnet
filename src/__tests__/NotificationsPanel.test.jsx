import React from 'react';
import { render } from '@testing-library/react';
import NotificationsPanel from '../components/Notifications/NotificationsPanel';

describe('NotificationsPanel', () => {
    test('matches snapshot', () => {
        const sampleNotifications = [
            { id: 1, type: 'comment', text: 'New comment', timestamp: new Date().toISOString() }
        ];
        const { container } = render(<NotificationsPanel notifications={sampleNotifications} />);
        expect(container).toMatchSnapshot();
    });
});
