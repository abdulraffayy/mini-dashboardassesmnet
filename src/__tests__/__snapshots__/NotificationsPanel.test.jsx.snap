// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsPanel matches snapshot 1`] = `
<div>
  <div
    class="bg-white p-8 rounded-xl shadow-md"
  >
    <h2
      class="text-xl font-semibold mb-6"
    >
      Notifications
    </h2>
    <ul>
      <li>
        <div
          class="flex items-start gap-4 pb-4"
        >
          <span
            class="bg-blue-400 rounded-lg flex items-center justify-center w-10 h-10"
          >
            <svg
              class="w-6 h-6 text-white"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 15a2 2 0 01-2 2H7l-4 4V5a2 2 0 012-2h14a2 2 0 012 2z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
          <div
            class="flex flex-col"
          >
            <span
              class="font-bold text-base text-gray-900"
            >
              New comment
            </span>
            <span
              class="text-xs text-gray-500 mt-1"
            >
              less than a minute ago
            </span>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
